src/script/descriptor.cpp

```diff
@@ -524,6 +540,21 @@ class RawDescriptor final : public DescriptorImpl
 public:
     RawDescriptor(CScript script) : DescriptorImpl({}, {}, "raw"), m_script(std::move(script)) {}
     bool IsSolvable() const final { return false; }
+
+    Optional<OutputType> GetOutputType() const override
+    {
+        CTxDestination dest;
+        ExtractDestination(m_script, dest);
```

Comment:
Assert result?

Reply:
Done

Reply:
Actually reverted that. It is possible, and perhaps expected, that we get a script which causes `ExtractDestination` to fail. Such a script would be a `CNoDestination`.